### ISSUE TITLE
unable to ssh in to my container

### ISSUE THREAD

**Issue by hesco** [weaveworks/weave#2350]
```
/usr/bin/docker run -d -e WEAVE_CIDR=10.0.1.16/29 --net=weave \
--name=samora007.example.com -h samora007.example.com\
 --memory=4g --restart=always -p 5432:5432 \
-v /data/etc/postgresql/samora007.example.com:/etc/postgresql \
-v /data/postgresql:/var/lib/postgresql \
-v /data/var/log/postgresql/samora007.example.com:/var/log/postgresql \
-v /data/home/ymdbackups:/home/ymdbackups samora007.example.com

ipam_ip=$(/usr/bin/docker inspect \
-f "{{ .NetworkSettings.Networks.weave.IPAddress }}" $DOMAIN)

weave_inspect_env=$(/usr/bin/docker inspect \
-f "{{ .Config.Env }}" $DOMAIN) 
```

with a little parsing and such gives me:

```
ipam: 10.0.2.1
weave: 10.0.1.16
```

But I can run it with an additional --net=bridge and get a docker0 ip, which I can use to ssh in, but then I get an lo and eth0, but no ethwe interfaces.  

```traceroute 10.0.1.16``` 

finds the weave IP expose'd to the local hardware, and then dies like this:

```
root@dessalines011:~/docker-helper-scripts/lib/docker/postgresql# traceroute 10.0.1.16 
traceroute to 10.0.1.16 (10.0.1.16), 30 hops max, 60 byte packets
 1  10.0.1.61 (10.0.1.61)  2996.995 ms !H  2996.938 ms !H  2996.922 ms !H
root@dessalines011:~/docker-helper-scripts/lib/docker/postgresql#
```

**Comment by bboreham** [comment]
`-e WEAVE_CIDR=10.0.1.16/29 --net=weave` suggests some mix-up.  The first part is intended for use with the Weave proxy, and the second is asking to use the Weave plugin.  As you observe, you have now got two addresses on the Weave network.  The next release of Weave Net will stop the proxy adding a second address under these circumstances (fixed in #2330).

If you don't want two addresses, remove one of the options.

If you do want two addresses, add the second one using `weave attach` or `docker network connect`.

Once you clean that up, if you're still having trouble can you give us some details of how the container ends up configured, e.g.:

    `docker exec mycontainer ip addr show`
    `docker exec mycontainer ip route`

When using the plugin, the syntax to ask for a particular ip address is `docker run ... --ip=xxx`, and you'd have to set up another Docker Network using the `weavemesh` driver with the right subnet. There is an example at #2046.

Incidentally `10.0.1.16/29` has a host number (the part of the address after the network prefix which is the first 29 bits) of zero.  This is not allowed as a host address by RFC 1122 (section 3.2.1.3), although that probably isn't a key part of your problem.

**Comment by rade** [comment]
The above analysis looks sound. Closing for now; if new information comes to light then please re-open.
